### PR TITLE
feat(provider/gcp): Add ability for default app creds to impersonate …

### PR DESCRIPTION
**Pull request by ttomsu**
…a service account.

### CODE REVIEW

File: clouddriver-google/clouddriver-google.gradle [hunk 1]
```diff
@@ -6,6 +6,9 @@ dependencies {
   compile spinnaker.dependency('frigga')
   compile spinnaker.dependency('bootActuator')
   compile spinnaker.dependency('bootWeb')
+
+  compile "com.google.apis:google-api-services-iam:v1-rev225-1.23.0"
```
Comment (duftler) [inline clouddriver-google/clouddriver-google.gradle]: Could leave a todo here to move these to spinnaker-dependencies when this stabilizes.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonateServiceAccountCredentials.groovy [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+package com.netflix.spinnaker.clouddriver.google.security
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
+import com.google.api.client.http.HttpTransport
+import com.google.api.client.json.JsonFactory
+import com.netflix.spinnaker.clouddriver.google.ComputeVersion
+import groovy.util.logging.Slf4j
+
+@Slf4j
+class GoogleImpersonateServiceAccountCredentials extends GoogleCredentials {
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonateServiceAccountCredentials.groovy]: Should this be 'Impersonated' for consistency with the other classes?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonatedCredential.groovy [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+   * instance's service account (aka application default credentials) to impersonate another service
+   * account by using the signJwt endpoint. The instance's service account must have the
+   * 'Service Account Token Creator' role.
+   *
+   * This use case differs slightly from the 'GoogleCredential.Builder().setServiceAccountUser.'
+   * With that setter method, the credential still requires the private key of the service account.
+   * Attempts to circumvent that requirement (by using reflection magic to set the field) resulted
+   * in a 401 Unauthorized response from the server.
+   *
+   * This only applies when running Clouddriver on GCP (where Application Default Credentials are
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonatedCredential.groovy]: What about when not running on GCP but specifying the `GOOGLE_APPLICATION_CREDENTIALS` env variable?

File: clouddriver-google/clouddriver-google.gradle (hunk 1 above)
Comment (ttomsu) [reply clouddriver-google/clouddriver-google.gradle]: Done.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonateServiceAccountCredentials.groovy (hunk 2 above)
Comment (ttomsu) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonateServiceAccountCredentials.groovy]: Done.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonatedCredential.groovy (hunk 3 above)
Comment (ttomsu) [reply clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleImpersonatedCredential.groovy]: Done.

**Pull request merged**